crates/league-modpkg/src/chunk.rs

```diff
@@ -0,0 +1,58 @@
+use std::{
+    borrow::Cow,
+    io::{BufReader, Read},
+};
+
+use byteorder::{ReadBytesExt as _, LE};
+use io_ext::ReaderExt as _;
+
+use crate::error::ModpkgError;
+
+#[derive(Debug, PartialEq, PartialOrd)]
+pub struct ModpkgChunk {
+    path: Cow<'static, str>,
+    path_hash: u64,
+    compressed_size: usize,
+    uncompressed_size: usize,
+    data_offset: usize,
+    checksum: u64,
+}
+
+impl ModpkgChunk {
+    pub fn read(reader: &mut BufReader<impl Read>) -> Result<Self, ModpkgError> {
+        let path = reader.read_len_prefixed_string::<LE>()?;
```

Comment:
This can and probably be stored in separate lookup table.
Having mod chunks be variable sized sucks for reading this all at once.